test/State.ts

```diff
@@ -0,0 +1,25 @@
+import { type BaseWallet, getBytes } from 'ethers'
+
+import { ecsign, toRpcSig, keccak256 as keccak256_buffer } from 'ethereumjs-util'
+import { type StateStruct } from '../typechain-types/contracts/Nitro-SCW.sol/NitroSmartContractWallet'
+
+export function signStateHash (stateHash: string, owner: BaseWallet, intermediary: BaseWallet): [string, string] {
+  const msg1 = Buffer.concat([
+    Buffer.from('\x19Ethereum Signed Message:\n32', 'ascii'),
+    Buffer.from(getBytes(stateHash))
+  ])
+
+  const ownerSig = ecsign(keccak256_buffer(msg1), Buffer.from(getBytes(owner.privateKey)))
+  const intermediarySig = ecsign(keccak256_buffer(msg1), Buffer.from(getBytes(intermediary.privateKey)))
+
+  return [toRpcSig(ownerSig.v, ownerSig.r, ownerSig.s), toRpcSig(intermediarySig.v, intermediarySig.r, intermediarySig.s)]
+}
+
+export function hashState (state: StateStruct): string {
+  throw new Error('TODO:Figure out how to get correct the hash of the state')
```

Comment:
This is kind of silly but I've been struggling to hash the state correctly 😢 . I think we want to use something like `ethers.solidityPackedKeccak256` but it doesn't seem to like the `tuple(uint256,bytes32,uint256)[]` type.

Reply:
I had a go at this over on #15 . Not sure if I was successful or not.